src/services/codefixes/fixImplicitThis.ts

```diff
@@ -0,0 +1,68 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "fixImplicitThis";
+    const errorCodes = [Diagnostics.this_implicitly_has_type_any_because_it_does_not_have_a_type_annotation.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const { sourceFile, program, span } = context;
+            let diagnostic: DiagnosticAndArguments | undefined;
+            const changes = textChanges.ChangeTracker.with(context, t => {
+                diagnostic = doChange(t, sourceFile, span.start, program.getTypeChecker());
+            });
+            return diagnostic ? [createCodeFixAction(fixId, changes, diagnostic, fixId, Diagnostics.Fix_all_implicit_this_errors)] : emptyArray;
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            doChange(changes, diag.file, diag.start, context.program.getTypeChecker());
+        }),
+    });
+
+    function doChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, pos: number, checker: TypeChecker): DiagnosticAndArguments | undefined {
+        const token = getTokenAtPosition(sourceFile, pos);
+        Debug.assert(token.kind === SyntaxKind.ThisKeyword);
+
+        const fn = getThisContainer(token, /*includeArrowFunctions*/ false);

```

Comment:
Could we be in a default argument of the function?  Does that matter?

Reply:
In a parameter initializer, 'this' refers to the same 'this' as inside the body of the function, not to 'this' of an outer function. So this will work just as well in that case.